src/commands/upload.ts

```diff
@@ -0,0 +1,303 @@
+
+    // Step 3: Instantiating the ActionOffchainRollup contract.
+    ux.action.start('Instantiating the ActionOffchainRollup contract')
+    const brickProfileAbi = await this.loadAbiByContractId(
+      registry,
+      brickProfileContractId
+    )
+    const brickProfileContractKey = await registry.getContractKeyOrFail(
+      brickProfileContractId
+    )
+    const brickProfile = new PinkContractPromise(
+      apiPromise,
+      registry,
+      brickProfileAbi,
+      brickProfileContractId,
+      brickProfileContractKey
+    )
+    const rollupAbi = new Abi(
+      await this.loadAbiByCodeHash(
+        isDev
+          ? '0xe0a086ccadbac348b625e859b46175224b226d29fa842f051e49c6fcc85dee62'
+          : '0x96ca5480eb52b8087b1e64cae52c75e6db037e1920320653584ef920db5d29d5'
+      )
+    )
+    const blueprint = new PinkBlueprintPromise(
+      apiPromise,
+      registry,
+      rollupAbi,
+      rollupAbi.info.source.wasmHash.toHex()
+    )
+
+    const result = await signAndSend<PinkBlueprintSubmittableResult>(
+      blueprint.tx.withConfiguration(
+        { gasLimit: 1000000000000 },
+        flags.rpc,
+        flags.consumerAddress,
+        fs.readFileSync(
+          buildAssets && buildAssets.length
+            ? upath.join(buildAssets[0].outputPath, buildAssets[0].name)
+            : upath.join(process.cwd(), 'dist', 'index.js'),
+          'utf8'
+        ),
+        flags.coreSettings || '',
+        brickProfileContractId
+      ),
+      pair
+    )
+    await result.waitFinalized()
+    const contractPromise = result.contract
+    ux.action.stop()
+    this.log(
+      'The ActionOffchainRollup contract has been instantiated:',
+      contractPromise.address.toHex()
+    )
+
+    // Step 4: Setting up the actions.
+    ux.action.start('Setting up the actions')
+    const { output: attestorQuery } =
+      await contractPromise.query.getAttestAddress(cert.address, { cert })
+    const attestor = attestorQuery.asOk.toHex()
+    const selectorUint8Array = rollupAbi.messages
+      .find((i) => i.identifier === 'answer_request')
+      ?.selector.toU8a()
+    const selector = Buffer.from(selectorUint8Array!).readUIntBE(
+      0,
+      selectorUint8Array!.length
+    )
+    const actions = [
+      {
+        cmd: 'call',
+        config: {
+          codeHash: rollupAbi.info.source.wasmHash.toHex(),
+          callee: contractPromise.address.toHex(),
+          selector,
+          input: [],
+        },
+      },
+      {
+        cmd: 'log',
+      },
+    ]
+    const { output: numberQuery } = await brickProfile.query.workflowCount<u16>(
+      pair.address,
+      { cert }
+    )
+    const num = numberQuery.asOk.toNumber()
+    const { blocknum: initBlockNum } = await registry.phactory.getInfo({})
+
+    await signAndSend(
+      brickProfile.tx.addWorkflow(
+        { gasLimit: 1000000000000 },
+        `My Phat Function ${numberQuery.asOk.toNumber()}`,
```

Comment:
Should call this `Phat Contract` or `Phala Oracle` instead?

Reply:
Let's use the term "Phat Contract" consistently across different codebases.